### PR TITLE
pull request for xpdAcq issues

### PR DESCRIPTION
**Pull request by chiahaoliu**
move plotting out from saving function

Need to refactor _feature_gen function later

### CODE REVIEW

File: xpdacq/analysis.py [hunk 1]
```diff
@@ -89,16 +90,14 @@ def _timestampstr(timestamp):
     #corrected_timestampstring = timestampstring.replace(':','-')
     return timestampstring
 
-def save_last_tif():
+def save_last_tiff():
```
Comment (tacaswell) [inline xpdacq/analysis.py]: In general, if you are going to do rename like this you should probably include

```python
save_last_tif = save_last_tiff
```

to make sure you do not break any code on the beamline computers.

Comment (chiahaoliu) [inline xpdacq/analysis.py]: good lesson, @tacaswell

**Comment by tacaswell** [comment]
You can keep pushing commits to this branch to update your PR.

**Comment by sbillinge** [comment]
@tacaswell I am still trying to understand our workflow.

If I understand correctly, MASTER in xpdacq/xpdacq now contains deployed code, and nothing should be pushed there until it has actually been tested in the production environment (in our case XPD).  Does this mean that we need a branch in xpdacq/xpdacq against we make pull requests, maybe we could call it ``ready-to-test`` or something.  Then we can test ``ready-to-test`` at XPD during a maintenance period.  If it works, then we can merge it to master. Then the workflow could be like this:

```
1) read-to-test is branched from MASTER on xpdacq/xpdacq
2) all developers fork their own copy of xpdacq/xpdacq which comes with two branches, MASTER and ready-to-test
3) we all clone myname/xpdacq
4) we make a local branch or branches for developing (do we branch off MASTER or RTT?)
5) we do lots of local testing for syntax errors and local stuff.  Ideally we have test cases and we make them pass, but we haven't learned how to put that into our workflow yet.  It would be good to start that for the V1.0 release milestone, but not before.
6) we periodically push our repo to our fork and request comments and feedback from others who we have linked to the fork.
7) When we are ready we make a pull request to the ``read-to-test`` branch of xpdacq
8) this is tested, modified then merged.

It seems like the git workflow is a bit easier than this (no ready-to-test branch) but then is there a way to test the version of the code in the pull request in the production environment before we merge it?

**Comment by sbillinge** [comment]
@tacaswell btw, I sent a couple of invitations to Dan to join this but he hasn't responded.  Does he not want to be looped in (fine with me if you are our point of contact in DAMA)?  Otherwise it would be good to get his feedback too.

**Comment by tacaswell** [comment]
What in the 'ready-to-test' branch?

**Comment by tacaswell** [comment]
Sigh, I should finish reading before commenting no?

**Comment by tacaswell** [comment]
It is a matter of naming, but most projects invert the names, 'master' is where everything gets merged into and every so often you tag a release and create a maintenance branch which only gets bug fixes.  In this case it matters far more what is _deployed_ to the production environment (which really should be a tagged version) than the state of any given branch.

This is also thing that automated testing should really be able to help with.

**Pull request merged**